### ISSUE TITLE
Support for `track_cluster_stats`

### ISSUE THREAD

**Issue by jrauschenbusch** [solo-io/gloo#9243]
### Gloo Edge Product

Open Source

### Gloo Edge Version

1.16.8

### Is your feature request related to a problem? Please describe.

I would like to get more details about traffic distribution via `per_endpoint_stats`.

In general i would like to understand load balancing behavior during rolling restarts of instances belonging to an upstream cluster. I think per endpoint metrics would help here.

- https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/cluster.proto#config-cluster-v3-cluster
- https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/cluster.proto#config-cluster-v3-trackclusterstats

Basically another thing would be to get the current load weight per host as metric. But don't know if there is support by envoy.

**Comment by AkshayAdsul** [comment]
Xero Team mainly interested in the request_response_sizes option.

**Comment by AkshayAdsul** [comment]
Thanks Duncan for removing the stale label.
Xero team has mentioned this would still be a good option to have. They'll discuss internally and let us know the priority and update the tracker.
